yarn-project/end-to-end/src/e2e_p2p/reex.test.ts

```diff
@@ -0,0 +1,80 @@
+import { type AztecNodeService } from '@aztec/aztec-node';
+import { type SentTx, sleep } from '@aztec/aztec.js';
+
+import { beforeAll, describe, it } from '@jest/globals';
+import fs from 'fs';
+
+import { createNodes } from '../fixtures/setup_p2p_test.js';
+import { P2PNetworkTest, WAIT_FOR_TX_TIMEOUT } from './p2p_network.js';
+import { submitComplexTxsTo } from './shared.js';
+
+const NUM_NODES = 4;
+const NUM_TXS_PER_NODE = 1;
+const BOOT_NODE_UDP_PORT = 41000;
+
+const DATA_DIR = './data/re-ex';
+
+describe('e2e_p2p_reex', () => {
+  let t: P2PNetworkTest;
+  beforeAll(async () => {
+    t = await P2PNetworkTest.create('e2e_p2p_reex', NUM_NODES, BOOT_NODE_UDP_PORT);
+
+    t.logger.verbose('Setup account');
+    await t.setupAccount();
+
+    t.logger.verbose('Deploy spam contract');
+    await t.deploySpamContract();
+
+    t.logger.verbose('Apply base snapshots');
+    await t.applyBaseSnapshots();
+
+    t.logger.verbose('Setup nodes');
+    await t.setup();
+  });
+
+  afterAll(async () => {
+    await t.teardown();
+    for (let i = 0; i < NUM_NODES; i++) {
+      fs.rmSync(`${DATA_DIR}-${i}`, { recursive: true, force: true });
+    }
+  });
+
+  it('validators should re-execute transactions before attesting', async () => {
+    // create the bootstrap node for the network
+    if (!t.bootstrapNodeEnr) {
+      throw new Error('Bootstrap node ENR is not available');
+    }
+
+    t.ctx.aztecNodeConfig.validatorReEx = true;
+
```

Comment:
I ran this test with the flag set to `false` and it still passes. How hard is it to assert that they actually re-execute?

Reply:
Good point, I can probably find a way to add a spy into the validator

Reply:
Aside: re ex is enabled by default on the nodes, so all of these tests will run with re execution, this test just adds doing something in public as the other tests lacked that

Reply:
One way to do it could probably be if we can just propose a block with an invalid transaction? Seeing that they catch it